src/Components/Search.js

```diff
@@ -30,19 +32,35 @@ export default function Search() {
   const [user, loading] = useAuthState(auth);
 
   const [localUser, setLocalUser] = useContext(LocalUserContext);
-  let [search, setSearch] = useState("");
+  let [search, setSearch] = useState(location.state ?? null);
 
   // To-Do: Handle loading/error states from API call
   const {
     data: searchResults,
-    loading: searchLoading,
+    isLoading: searchLoading,
+    isFetching: searchFetching,
     error: apiError,
-  } = useAPISearch(search);
+  } = useAPISearch(search, 10);
+
+  const loadingSearch = searchLoading || searchFetching;
+
+  const [indexOfUsers, setIndexOfUsers] = useState();
 
   useEffect(() => {
     setSearch(location.state);
   }, [location]);
 
+  // Locates where to place the "USERS" banner
+  useEffect(() => {
```

Comment:
Clever!  

If you are using `useEffect` here just to make a computation only when a variable changes, you could also consider `useMemo`:

```js
const indexOfUsers = useMemo(() => {
  for (let i = 0; i < searchResults.length; i++) {
    if (searchResults[i].uid) {
      return i;
    }
  }
  return undefined;
}, [searchResults]);
```

The function will only be run when searchResults changes.  Otherwise, `useMemo` will have stored the previous result and will just return that.

Reply:
Ahh, that's much better!! Great suggestion!